My first Blog post (#74)
scientific-python/blog.scientific-python.org: content/posts/networkx/GSoC-2022/index.md

---
title: "Hello World"
date: 2022-06-09
draft: false
description: "Getting started"
tags: ["gsoc", "networkx"]
displayInList: true
author: ["kpetridis24"]

resources:
  - name: featuredImage

>     src: "GSoC_logo.png"

Reviewer: @mjschwenne had created a version of the logo that has both the networkx & GSoC logos in a single png - I think we should use that here as well --- esp. since the blog has posts from GSoC participants from other projects.

The easiest way to do this is probably to copy the `mjschwenne_GSoC.png` file over from one of the other directories (you can rename it in this folder if you'd like). Even better might be to try to link out to it, but we can always worry about that in a followup PR.

Reviewer 2: I actually got the idea from the other GSoC posts like Aitik Gupta's Matplotlib posts. All I did was fire up GIMP and replace the Matplotlib logo with a NetworkX logo :smile:.

    params:
      description: "Google Summer of Code Logo with NetworkX logo"
      showOnTop: true
---
 
I got accepted as a GSoC contributor, and I am so excited to spend the summer working on 
such an incredibly interesting project. The mentors are very welcoming, communicative, fun 

> to be around, and I really look forward to collaborate with them.

Reviewer: This is a nice intro :) IMO it'd be good to include a sentence or two about what you'll be working on. You don't have to go into great detail or anything, but at least a mention of VF2++ and (if you're comfortable doing so) perhaps linking out to your GSoC proposal.

---
title: "GSoC 2022"
date: 2022-06-09
draft: false
description: "Getting started"
tags: ["gsoc", "networkx"]
displayInList: true
author: ["kpetridis24"]

resources:
  - name: featuredImage

>     src: "mjschwenne_GSoC.png"

Reviewer 3: Please use a tool like `pngcrunsh` to compress the image. Looking at it, it should be easy to get well bellow 100kb.

Author: @tupui Thanks for the feedback. I added a significant ammount of information. Check it out and if there is more that can go into it, let me know.

to verify/reject the isomorphism between two graphs, we examine every possible candidate pair of nodes (one from the first
and one from the second graph) and check whether going deeper into the DFS tree is feasible using specific rules. In case 
of feasibility establishment, the DFS tree is expanded, investigating deeper pairs. When one pair is not feasible, we 
go up the tree and follow a different brach, just like in a regular **DFS**. More details about the algorithm can be 
found [here](https://www.sciencedirect.com/science/article/abs/pii/S0166218X18300829). 

# Motivation
The major reasons I chose this project emanate from both my love for **Graph Theory**, and the fascinating nature of 
this individual project. The algorithm itself is so recent, that **NetworkX** is possibly going to hold one of the first
implementations of it. This might become a reference that is going to help to further develop and optimize future 
implementations of the algorithm by other organisations. Regarding my personal gain, I will become more familiar with 
the open source communities and their philosophy, I will collaborate with highly skilled individuals and cultivate a 

> significant ammount of experience on researching, working as a team, getting feedback and help when needed, contributing 

Author: @MridulS @tupui I added a section about me and included links where i considered it to be necessary. Give it a look, and if you want me to sorten the algorithm section and make it more general, let me know!

---
title: "GSoC 2022"
date: 2022-06-09
draft: false
description: "Getting started"
tags: ["gsoc", "networkx"]
displayInList: true
author: ["kpetridis24"]

resources:
  - name: featuredImage
    src: "GSoC_networkx_logo.png"
    params:
      description: "Google Summer of Code Logo with NetworkX logo"
      showOnTop: true
---

# Intro
I got accepted as a **GSoC** contributor, and I am so excited to spend the summer working on 
such an incredibly interesting project. The mentors are very welcoming, communicative, fun 
to be around, and I really look forward to collaborating with them. My application for GSoC 2022 
can be [found here](https://summerofcode.withgoogle.com/programs/2022/projects/V1hY83XG).

> # About me

Reviewer 3: Good update thank you. I would also add a link to your GitHub profile since it would be very relevant for GSoC.

My name is Konstantinos Petridis, and I am an **Electrical Engineering** student at the Aristotle University of 
Thessaloniki. I am currently on my 5th year of studies, with a **Major in Electronics & Computer Science**. Although a 
wide range of scientific fields fascinate me, I have a strong passion for **Computer Science**, **Physics** and **Space**.
I love to study, learn new things and don't hesitate to express my curiosity by asking a bunch of questions to the point 

> of being annoying :).  

Reviewer 3: A bit neutral, but this is not rendered. You can add a real smiley as we should support utf8 chars.

# Project
The project I'll be working on, is the implementation of **VF2++**, a state-of-the-art algorithm used for the 
[**Graph Isomorphism**](https://en.wikipedia.org/wiki/Graph_isomorphism) problem, which lies in the 
[complexity class](https://en.wikipedia.org/wiki/Complexity_class) [**NP**](https://en.wikipedia.org/wiki/NP_(complexity)). 

> The functionality of the algorithm reminds a regular, but more complex form of a 
> [**DFS**](https://en.wikipedia.org/wiki/Depth-first_search), but performed on the possible solutions rather than the graph 
> nodes. In order to verify/reject the isomorphism between two graphs, we examine every possible candidate pair of nodes 

Reviewer: ```suggestion
The functionality of the algorithm is similar to a regular, but more complex form of a 
[**DFS**](https://en.wikipedia.org/wiki/Depth-first_search), but performed on the possible solutions rather than the graph 
nodes. In order to verify/reject the isomorphism between two graphs, we examine every possible candidate pair of nodes 
```

(one from the first and one from the second graph) and check whether going deeper into the DFS tree is feasible using 
specific rules. In case of feasibility establishment, the DFS tree is expanded, investigating deeper pairs. When one pair 
is not feasible, we go up the tree and follow a different branch, just like in a regular **DFS**. More details about the 

> algorithm can be found [here](https://www.sciencedirect.com/science/article/abs/pii/S0166218X18300829). 

Reviewer 3: Always prefer the use of a DoI.
```suggestion
algorithm can be found [here](https://doi.org/10.1016/j.dam.2018.02.018). 
```

---
title: "GSoC 2022: NetworkX VF2++ Implementation"
date: 2022-06-09 
draft: false 

> description: "Getting started"

Reviewer 3: Can you write something meaningful here? Sorry if I missed this one.

I am planning on merging this today if there are no more comments.

Author: Do you refer to the description?

Reviewer 3: yes

Reviewer 3: It's a tweet like description

Author: It's the first time i create a blog so i was not familiar with the philosophy. I promise that the next blogs to come will be a lot more complete and will not require such long time to review. :)

Reviewer 3: No problem at all 😃. It's part of the learning and we are also learning our way here as we build this blog.

Reviewer 3: Seems like you have some formatting issues. Cf CI. For next time, you can install pre-commit hooks to format automatically for you (I should update our guidelines to point this out)

---

> title: "GSoC 2022"

Reviewer 3: ```suggestion
title: "GSoC 2022: NetworkX VF2++ Implementation"
```

date: 2022-06-09
draft: false
description: "Getting started"
tags: ["gsoc", "networkx"]
displayInList: true

> author: ["kpetridis24"]

Reviewer 3: ```suggestion
author: ["Konstantinos Petridis"]
```

resources:

- name: featuredImage
  src: "GSoC_networkx_logo.png"
  params:
  description: "Google Summer of Code Logo with NetworkX logo"
  showOnTop: true

---

# Intro

I got accepted as a **GSoC** contributor, and I am so excited to spend the summer working on
such an incredibly interesting project. The mentors are very welcoming, communicative, fun
to be around, and I really look forward to collaborating with them. My application for GSoC 2022
can be [found here](https://summerofcode.withgoogle.com/programs/2022/projects/V1hY83XG).

# About me

My name is Konstantinos Petridis, and I am an **Electrical Engineering** student at the Aristotle University of
Thessaloniki. I am currently on my 5th year of studies, with a **Major in Electronics & Computer Science**. Although a

> wide range of scientific fields fascinate me, I have a strong passion for **Computer Science**, **Physics** and **
> Space**.

Reviewer 3: ```suggestion
wide range of scientific fields fascinate me, I have a strong passion for **Computer Science**, **Physics** and **Space**.
```

I love to study, learn new things and don't hesitate to express my curiosity by asking a bunch of questions to the point

> of being annoying. You can find me on GitHub, [here](https://github.com/kpetridis24).

Reviewer 3: ```suggestion
of being annoying. You can find me on GitHub, [@kpetridis24](https://github.com/kpetridis24).
```

# Project

The project I'll be working on, is the implementation of **VF2++**, a state-of-the-art algorithm used for the
[**Graph Isomorphism**](https://en.wikipedia.org/wiki/Graph_isomorphism) problem, which lies in the

> [complexity class](https://en.wikipedia.org/wiki/Complexity_class) [**
> NP**](https://en.wikipedia.org/wiki/NP_(complexity)).

Reviewer 3: ```suggestion
[complexity class](https://en.wikipedia.org/wiki/Complexity_class) [**NP**](https://en.wikipedia.org/wiki/NP_(complexity)).
```
